src/app-layer-htp.c

```diff
@@ -2146,7 +2148,23 @@ static int HTPCallbackDoubleDecodeQuery(htp_tx_t *tx)
         return HTP_OK;
 
     uint64_t flags = 0;
-    htp_urldecode_inplace(tx->cfg, HTP_DECODER_URLENCODED, tx->parsed_uri->query, &flags);
+    size_t prevlen = bstr_len(tx->parsed_uri->query);
+    htp_status_t res = htp_urldecode_inplace(tx->cfg, HTP_DECODER_URLENCODED, tx->parsed_uri->query, &flags);
+    // shorter string means that uri was encoded
+    if (res == HTP_OK && prevlen > bstr_len(tx->parsed_uri->query)) {
+        HtpTxUserData *htud = (HtpTxUserData *) htp_tx_get_user_data(tx);
+        if (likely(htud == NULL)) {
+            htud = HTPMalloc(sizeof(*htud));
+            if (unlikely(htud == NULL))
+                return HTP_ERROR;
```

Comment:
we can probably be more forgiving here, instead of making the callback fail with an error simply don't set an event

Reply:
ok, I hesitated about this one